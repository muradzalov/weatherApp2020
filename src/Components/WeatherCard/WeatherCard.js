import React from 'react'

export default function WeatherCard(props) {


  const { humidity, temperature, weatherDescription, country, localTime, weatherIcon, city } = props.weatherInformation;

  return (
    // { Object.keys(props.weatherInformation).length }

    <div style={{
      display: 'flex',
      flexDirection: 'column',
      alignItems: 'center',
      marginBottom: '16px',
      lineHeight: '32px',
      letterSpacing: '1.8px',
      fontWeight: 'bold'
    }}>

      < div >
        <img alt="Weather icon" src={weatherIcon} />
      </div >

      <div style={{
        display: "flex",
        justifyContent: "flex-start",
        alignItems: "center",
        fontSize: '20px'
      }}>
        <div>
          <div>
            Humidity:
        </div>

          <div>
            Temperature:
        </div>

          <div>
            Weather Description:
        </div>

          <div>
            Location:
        </div>

          <div>
            Local Time:
        </div>
        </div>

        <div style={{
          display: "flex",
          alignItems: "flex-end",
          flexDirection: 'column',
          fontSize: '20px'
        }}>
          <div>
            {humidity && humidity}%
          </div>

          <div>
            {temperature && temperature}Â°F
          </div>

          <div>
            {weatherDescription && weatherDescription}
          </div>

          <div>
            {city && city}, {country && country}
          </div>

          <div>
            {localTime && (new Date(localTime).toLocaleString())}
          </div>
        </div>
      </div>

    </div>
  )
}
