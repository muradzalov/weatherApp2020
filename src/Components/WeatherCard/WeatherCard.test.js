import WeatherCard from '../WeatherCard/WeatherCard'
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from "enzyme-adapter-react-16";
 
Enzyme.configure( {adapter: new Adapter() });

describe('Testing the WeatherCard component', () => {
  
  const props = {
    weatherInformation: {
      humidity: 10,
      temperature: 20,
      weatherDescription: 'It is warm',
      country: 'USA',
      localTime: '7:22 PM',
      weatherIcon: ''
    }
  }
  
  it('renders a component', () => {
    const wrapper = shallow(<WeatherCard {...props} />);
    expect(wrapper.length).toBe(1);
  })

  const container = shallow(<WeatherCard {...props}/>);
  it('should match the snapshot', () => {
    expect(container.html()).toMatchSnapshot();
  });
})

const sum = (a, b) => {
  return a + b;
}

test('4 + 4 using our sum function should equal 8', () => {
  expect(sum(4, 4)).toBe(8);
})
