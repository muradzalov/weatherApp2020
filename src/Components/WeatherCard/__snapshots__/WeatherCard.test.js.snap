// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the WeatherCard component should match the snapshot 1`] = `"<div style=\\"display:flex;flex-direction:column;align-items:center;margin-bottom:16px;line-height:32px;letter-spacing:1.8px;font-weight:bold\\"><div><img alt=\\"Weather icon\\" src=\\"\\"/></div><div style=\\"display:flex;justify-content:flex-start;align-items:center;font-size:20px\\"><div><div>Humidity:</div><div>Temperature:</div><div>Weather Description:</div><div>Country:</div><div>Local Time:</div></div><div style=\\"display:flex;align-items:flex-end;flex-direction:column;font-size:20px\\"><div>10</div><div>20</div><div>It is warm</div><div>USA</div><div>Invalid Date</div></div></div></div>"`;
